packages/api/src/routers/tab.ts

```diff
@@ -1,59 +1,66 @@
 import { z } from "zod";
-import { createTRPCRouter, publicProcedure } from "../trpc";
+import { TRPCError, createTRPCRouter, publicProcedure } from "../trpc";
+import { prismaErrorHandler } from "../utils";
 
 export const tabRouter = createTRPCRouter({
   getTab: publicProcedure
     .input(
       z.object({
-        user1ID: z.string(),
-        user2ID: z.string(),
+        debtorId: z.string(),
+        creditorId: z.string(),
       })
     )
     .query(async ({ input, ctx }) => {
-      const tab = await ctx.prisma.tab.findFirst({
-        where: {
-          debtorID: input.user1ID,
-          creditorID: input.user2ID,
-        },
-      });
-
-      if (tab?.amount != 0) {
-        return tab?.amount;
-      } else {
-        const inverseTab = await ctx.prisma.tab.findFirst({
+      try {
+        const tab = await ctx.prisma.tab.findFirstOrThrow({
           where: {
-            debtorID: input.user2ID,
-            creditorID: input.user1ID,
+            debtorID: input.debtorId,
+            creditorID: input.creditorId,
           },
         });
-        if (inverseTab?.amount == 0 || inverseTab?.amount == undefined) {
-          return 0;
-        }
 
-        return -inverseTab?.amount;
+        return tab.amount;
+      } catch (e) {
+        if (e instanceof Error) {
+          prismaErrorHandler(e);
+        }
       }
     }),
-  getAllTabs: publicProcedure
+
+  getAllTabsForAccount: publicProcedure
     .input(
       z.object({
         userID: z.string(),
       })
     )
     .query(async ({ input, ctx }) => {
-      const i_owe_u = await ctx.prisma.tab.findMany({
-        where: {
-          debtorID: input.userID,
-        },
-      });
+      try {
+        const all_tabs = await ctx.prisma.tab.findMany({
+          where: {
+            OR: [{ debtorID: input.userID }, { creditorID: input.userID }],
+          },
+        });
 
-      const u_owe_me = await ctx.prisma.tab.findMany({
-        where: {
-          creditorID: input.userID,
-        },
-      });
+        if (all_tabs.length == 0) {
+          throw new TRPCError({
```

Comment:
Not sure if this should actually be an error or if we should just return a empty list; thoughts @RoryMcDermid @ashleendaly @pkitazos ?